docs/examples/EKS.md

```diff
@@ -177,3 +178,25 @@ spec:
 ```
 
 if you do not provide these fields, a role will be created for your instance-group by the controller (will require IAM access).
+
+
+### Auto-Updating AWS EKS Optimised AMI ID
```

Comment:
We may want to add a comment that there might be some delay due to reconcile not being triggered.
E.g. new AMI is out, TTL for GetParameter is expired so we will get the new AMI, but nothing is triggering the reconcile. 
Until the new reconcile is triggered upgrade will not happen.

This is good if you use the locking mechanism, because removing the lock annotation will trigger the reconcile, but otherwise it might take a while until that happens.

In addition we need to add the newly added AWS SSM permissions to documentation here:
https://github.com/keikoproj/instance-manager/blob/master/docs/INSTALL.md

Reply:
> We may want to add a comment that there might be some delay due to reconcile not being triggered.
> E.g. new AMI is out, TTL for GetParameter is expired so we will get the new AMI, but nothing is triggering the reconcile.

I thought kubernetes objects get periodically reconciled?

Reply:
They do, but I think the default kubebuilder period is 24hrs, so that would be significant delay.
There are other thing that can trigger reconcile such as spot recommendations etc.

Reply:
from: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/config/v1alpha1#ControllerConfigurationSpec
```
// SyncPeriod determines the minimum frequency at which watched resources are
	// reconciled. A lower period will correct entropy more quickly, but reduce
	// responsiveness to change if there are many watched resources. Change this
	// value only if you know what you are doing. Defaults to 10 hours if unset.
	// there will a 10 percent jitter between the SyncPeriod of all controllers
	// so that all controllers will not send list requests simultaneously.
	// +optional
	SyncPeriod *metav1.Duration `json:"syncPeriod,omitempty"`
```

Reply:
That's it - we don't set it, so have the 10hr default.
Wonder if we should make this a global controller flag (we can do that later if we want), but would be good to add a small comment in the docs on this if this is the expected behavior (may take up to 10hrs + API TTL to get upgrade trigger)